src/test/cypress/integration/quiz-exercise/QuizExerciseAssessment.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { artemis } from '../../support/ArtemisTesting';
+import shortAnswerQuizTemplate from '../../fixtures/quiz_exercise_fixtures/shortAnswerQuiz_template.json';
+import multipleChoiceQuizTemplate from '../../fixtures/quiz_exercise_fixtures/multipleChoiceQuiz_template.json';
+
+// Accounts
+const admin = artemis.users.getAdmin();
+const tutor = artemis.users.getTutor();
+const student = artemis.users.getStudentOne();
+
+// Requests
+const courseManagementRequest = artemis.requests.courseManagement;
+
+// Common primitives
+let course: any;
+let quizExercise: any;
+
+describe('Quiz Exercise Assessment', () => {
+    before('Set up course', () => {
+        cy.login(admin);
+        courseManagementRequest.createCourse().then((response) => {
+            course = response.body;
+            courseManagementRequest.addStudentToCourse(course.id, student.username);
+            courseManagementRequest.addTutorToCourse(course, tutor);
+        });
+    });
+
+    afterEach('Delete Quiz', () => {
+        deleteQuiz();
+    });
+
+    after('Delete Course', () => {
+        cy.login(admin);
+        courseManagementRequest.deleteCourse(course.id);
+    });
+
+    describe('MC Quiz assessment', () => {
+        before('Creates a quiz and a submission', () => {
+            createQuiz();
+        });
+
+        it('Assesses a mc quiz submission automatically', () => {
+            cy.login(student);
+            courseManagementRequest.startExerciseParticipation(course.id, quizExercise.id);
+            courseManagementRequest.createMultipleChoiceSubmission(quizExercise, [0, 2]);
+            cy.wait(5000);
+            cy.visit('/courses/' + course.id + '/exercises/' + quizExercise.id);
+            cy.contains(quizExercise.title);
+            cy.contains('Score');
+        });
+    });
+
+    describe('SA Quiz assessment', () => {
+        before('Creates a quiz and a submission', () => {
+            createQuiz(shortAnswerQuizTemplate);
+        });
+
+        it('Assesses a sa quiz submission automatically', () => {
+            cy.login(student);
+            courseManagementRequest.startExerciseParticipation(course.id, quizExercise.id);
+            courseManagementRequest.createShortAnswerSubmission(quizExercise, ['give', 'let', 'run', 'desert', 'cry', 'goodbye']);
+            cy.wait(5000);
+            cy.visit('/courses/' + course.id + '/exercises/' + quizExercise.id);
+            cy.contains(quizExercise.title);
+            cy.contains('Score');
+        });
+    });
+});
+
+function createQuiz(quizQuestions: any = multipleChoiceQuizTemplate) {
+    courseManagementRequest.createQuizExercise({ course }, [quizQuestions], undefined, undefined, 1).then((quizResponse) => {
```

Comment:
I am a little bit confused about the duration = 1. I assume the parameter is a number of seconds. So you create a quiz with the release date = now and a duration of 1 second. How is it possible that the tests above are still able to participate in the exercise and make submissions? The 1 second should be over when they start participating or do I misunderstand something?

Reply:
The quiz duration is one second because the submission is created via api request. That doesnt usually take more than a second. Also the server only starts assessing the submissions after the duration so to keep the tests as fast as possible i chose the smallest possible duration.

Reply:
Well yeah, but the test then only has a 1 second timeframe to do multiple things.
1. You create the quiz exercise with a duration of 1
2. You start the quiz now (from here the test has 1 second to do the submission)
3. The test needs to send a login request and wait for its response
4. The test sends the quiz participation request and waits for its response
4. The test sends the actual submission to the student (this request needs to reach the server within that initial 1 second timeframe)
So after starting the exercise and before the 1 second timeframe is over Cypress needs to do 2 requests in sequence and wait for their response and then send the submission request, which also has to reach the backend in that 1 second timeframe.
If the testserver is under heavy load oder the buildagent isn't as fast as others this test might be very flaky. Did you run this test multiple times and check if it is flaky? Because to me it seems that it should fail most of the time, unless the backend allows the late submission

Reply:
i ran the pipeline multiple times and also many many times locally and it never failed for this reason is all i can say. If you really worry about this so much i can increase it to two seconds.